source/simulator/nullspace.cc

```diff
@@ -446,6 +446,9 @@ namespace aspect
             {
               fe[introspection.extractors.pressure].get_function_values (relevant_dst, in.pressure);
               fe[introspection.extractors.temperature].get_function_values (relevant_dst, in.temperature);
+
+              in.velocity=velocities;
+
```

Comment:
I think it might be easier to just get rid of the local `velocities` array altogether and simply always use `in.velocity` instead.